blockchain/wallet/wallet.go

```diff
@@ -48,21 +53,24 @@ type Wallet struct {
 	chain          *protocol.Chain
 	rescanProgress chan struct{}
 	ImportPrivKey  bool
+	ImportKeyLock  sync.Mutex
 	keysInfo       []KeyInfo
 }
 
 //NewWallet return a new wallet instance
-func NewWallet(walletDB db.DB, account *account.Manager, asset *asset.Registry, chain *protocol.Chain) (*Wallet, error) {
+func NewWallet(walletDB db.DB, account *account.Manager, asset *asset.Registry, chain *protocol.Chain, xpubs []pseudohsm.XPub) (*Wallet, error) {
```

Comment:
func NewWallet(walletDB db.DB, account *account.Manager, asset *asset.Registry, chain *protocol.Chain, xpubs []pseudohsm.XPub) (*Wallet, error) {
		w := &Wallet{
			DB:             walletDB,
			AccountMgr:     account,
			AssetReg:       asset,
			chain:          chain,
			rescanProgress: make(chan struct{}, 10),
			keysInfo:       make([]KeyInfo, 0),
		}
		if err := w.loadWalletInfo(); err != nil {
			return nil, err
		}
		if err := w.loadKeysInfo(xpubs); err != nil {
			return nil, err
		}
		go w.walletUpdater()
		return w, nil
	}	
	

	func (w *Wallet) loadKeysInfo(xpubs []pseudohsm.XPub) error {
		if rawKeyInfo := w.DB.Get(privKeyKey); rawKeyInfo != nil {
			return json.Unmarshal(rawKeyInfo, &w.keysInfo)
		}
		if w.status.BestHeight == 0{
			return nil
		}
		for i, v := range xpubs {
	 		w.ImportAccountXpubKey(i, v, RecoveryIndex)
	 	}
	 	return nil
	}